lib/notifications/notification.ex

```diff
@@ -116,6 +116,8 @@ defmodule Notifications.Notification do
 
     {:ok, {source, db_record}} =
       Skate.Repo.transaction(fn ->
+        Skate.Repo.query!("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE")
```

Comment:
So if I understand [the Postgres docs](https://www.postgresql.org/docs/9.5/transaction-iso.html), the serializable isolation level will cause the transaction attempting to insert a duplicate notification to fail, which, if I'm not mistaken, will cause `Repo.transaction` to return `{:error, stuff}` rather than `{:ok, stuff}`. Currently that will cause a crash, but I think we should basically treat it the same as if the query below returns an `existing_record`. Does that make sense, or am I missing something?

Reply:
I learned a number of interesting things about Postgres isolation levels this morning. Two of my takeaways are this:

1. It's probably sufficient to just use the REPEATABLE READ isolation level here, since it seems like what we want to avoid is the "phantom read" phenomenon.
2. Postgres' SERIALIZABLE is not necessarily reliable due to a known bug, fixed in Postgres' master but not yet released. See https://jepsen.io/analyses/postgresql-12.3. 

I'm not clear from that paper whether the buggy behavior is a practical problem in our case, but it seems plausible, and if we can sidestep it by using a less rigorous serialization level that seems like a win-win.

Reply:
Interesting! Yeah, a lot of this stuff is new to me so I'm learning as I read, but I agree that repeatable read sounds like essentially what we want.

The Postgres docs do still say:
> Applications using this level must be prepared to retry transactions due to serialization failures.

Does that have any implications for how we want to structure the Elixir code?

Reply:
Now I'm reading through these docs again and perhaps REPEATABLE READ isn't strong enough? Will do more research.

Reply:
It might be worth revisiting the idea of using a nested `SELECT` inside of an `INSERT` and using a `WHERE NOT EXISTS` clause under that. I _think_ it would achieve the same thing, and it may be a pain to implement via Ecto, but there's a possibility it would be easier than this? Just a thought, I'm fine with either approach if we can make it work.

Reply:
One thing we could do is write a DB trigger on insert into bridge_movements that raises an error if the blackout period is violated, but if bumping the isolation level is like using a sledgehammer to swat a fly, writing a DB trigger feels like using a hydraulic press.

Reply:
Haha yeah that seems like way too much.

Reply:
Yeah I did an experiment with REPEATABLE READ and a couple of Postgres consoles and it's not strong enough. It's going to have to be SERIALIZABLE and we'll have to figure out the serialization anomaly question.

Reply:
Huh, interesting, well good to know. For what it's worth I think the serialization anomaly shouldn't be _too_ hard to handle, but it might involve some slightly annoying rearranging of the Elixir code.